src/main/java/org/jenkinsci/plugins/gitclient/JGitAPIImpl.java

```diff
@@ -2792,18 +2792,19 @@ private StoredConfig getConfig(String GIT_DIR) throws GitException {
                 /* Prefer peeled ref if available (for tag commit), otherwise take first tag reference seen */
                 String tagName = entry.getKey();
                 Ref tagRef = entry.getValue();
-                if (!entry.getValue().isPeeled()) {
+                if (!tagRef.isPeeled()) {
```

Comment:
Mild goldplating to get more mileage out of locals `tagRef` and `tagName` below. If it's too much I can back these out.